import "goby/protobuf/dccl_option_extensions.proto";

// required fields will be filled in for you by ApplicationBase
// if you choose not to do so yourself
message Header
{
  //
  // time
  //

  // result of goby::util::as<std::string>(goby_time())
  // e.g. "2002-01-20 23:59:59.000"
  required string time = 10 [(dccl.codec)="_time",
                             (dccl.in_head)=true];

  //
  // source
  //
  required string source_platform = 11 [(dccl.codec)="_platform<->modem_id",
                                        (dccl.max)=31,
                                        (dccl.min)=0,
                                        (dccl.in_head)=true];
  optional string source_app = 12 [(dccl.omit)=true];
  
  //
  // destination
  //
  enum PublishDestination { PUBLISH_SELF = 1; PUBLISH_OTHER = 2; PUBLISH_ALL = 3; }
  optional PublishDestination dest_type = 13 [default = PUBLISH_SELF];
  
  optional string dest_platform = 14 [(dccl.codec)="_platform<->modem_id",
                                      (dccl.max)=31,
                                      (dccl.min)=0,
                                      (dccl.in_head)=true]; // required if dest_type == other

}
